java/src/main/java/ai/rapids/cudf/NativeDepsLoader.java

```diff
@@ -158,6 +158,24 @@ public static void loadNativeDeps(String[] loadOrder, boolean preserveDeps) thro
     }
   }
 
+  /**
+   * Optionally load native dependencies. This method attempts to load the specified libraries
+   * but does not throw exceptions on failure. Instead, it returns true if all libraries were
+   * loaded successfully, false otherwise.
+   * @param loadOrder the base name of the libraries. For example libfoo.so would be passed in as
+   *                  "foo". The libraries are loaded in the order provided.
+   * @return true if all libraries were loaded successfully, false otherwise
+   */
+  public static boolean loadOptionalNativeDeps(String[] loadOrder) {
+    try {
+      loadNativeDeps(loadOrder, preserveDepsAfterLoad);
```

Comment:
so loadNativeDeps tries to find the file in [os]/[arch] subdirectories inside of the jar. How does that work with a library that comes from the system?

Reply:
Well in this case it would just find `libnvmljni.so` in the specified subdirectory, but then that lib would in turn use the standard mechanisms eg. `LD_LIBRARY_PATH` to find `libnvidia-ml.so.1`